packages/rum-core/src/domain/rumEventsCollection/view/trackInteractionToNextPaint.ts

```diff
@@ -0,0 +1,142 @@
+ * - if an entry with the same interaction id exists and but its duration is lower than the new one, then replace it in the list of worst interactions
+ */
+function processEntry(newInteraction: RumEventTiming | RumFirstInputTiming) {
+  const interaction = find(
+    longestInteractions,
+    (interaction) => newInteraction.interactionId === interaction.interactionId
+  )
+  const minLongestInteraction = longestInteractions[longestInteractions.length - 1]
+
+  if (interaction) {
+    if (newInteraction.duration > interaction.duration) {
+      const index = longestInteractions.indexOf(interaction)
+      longestInteractions[index] = newInteraction
+      sortAndTrimLongestInteractions()
+    }
+  } else if (
+    longestInteractions.length < MAX_INTERACTION_ENTRIES ||
+    newInteraction.duration > minLongestInteraction.duration
+  ) {
+    longestInteractions.push(newInteraction)
+    sortAndTrimLongestInteractions()
+  }
+}
+
+/**
+ * Compute the p98 longest interaction.
+ * For better performance the computation is based on 10 longest interactions and the interaction count of the current view.
+ */
+function estimateP98LongestInteraction(getInteractionCount: () => number) {
+  const interactionIndex = Math.min(longestInteractions.length - 1, Math.floor(getInteractionCount() / 50))
+  return longestInteractions[interactionIndex]
+}
+
+function sortAndTrimLongestInteractions() {
+  longestInteractions.sort((a, b) => b.duration - a.duration).splice(MAX_INTERACTION_ENTRIES)
+}
+
+function matchInteraction(candidate: RumEventTiming | RumFirstInputTiming) {
+  return longestInteractions.some(
+    (interaction) => candidate.duration === interaction.duration && candidate.startTime === interaction.startTime
+  )
+}
+
+let previousInteractionCount: number
+function trackInteractionCount(viewLoadingType: ViewLoadingType) {
+  if ('interactionCount' in performance) {
+    previousInteractionCount = (window as BrowserWindow).performance.interactionCount!
+    return {
+      getInteractionCount: () =>
+        viewLoadingType === ViewLoadingType.INITIAL_LOAD
+          ? (window as BrowserWindow).performance.interactionCount!
+          : (window as BrowserWindow).performance.interactionCount! - previousInteractionCount,
+      stop: noop,
+    }
+  }
+
+  let interactionCount = 0
+  let minKnownInteractionId = Infinity
+  let maxKnownInteractionId = 0
+
+  const observer = new (window as BrowserWindow).PerformanceObserver((entries: PerformanceObserverEntryList) => {
+    entries.getEntries().forEach((e) => {
+      const entry = e as unknown as RumEventTiming
+      if (entry.interactionId) {
+        minKnownInteractionId = Math.min(minKnownInteractionId, entry.interactionId)
+        maxKnownInteractionId = Math.max(maxKnownInteractionId, entry.interactionId)
+
+        interactionCount = maxKnownInteractionId ? (maxKnownInteractionId - minKnownInteractionId) / 7 + 1 : 0
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏what does `7` represent?

Reply:
From the [original PR from Google](https://github.com/GoogleChrome/web-vitals/pull/213): "interactionCount is an estimate based on a convention specific to Chrome". I found no more information